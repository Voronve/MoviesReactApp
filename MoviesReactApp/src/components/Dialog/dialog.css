.dialogWindow {
    font: 20px 'Montserrat', sans-serif;
    background-color: #424242;
    position: fixed;
    width: 60%;
    top: 10%;
    left: 20%;
    z-index: 2;
    padding-bottom: 50px;
  }

  .dialogWindow .dialogHeader {
    color: white;
    text-align: center;
  }

  .dialogWindow .crossButton, .dialogWindow a{
    cursor: pointer;
    color: red;
    border: none;
    outline: none;
    appearance: none;
    box-shadow: none;
    background-color: transparent;
    min-width: 10px;
    position: absolute;
    top: 10px;
    right: 10px;
  }