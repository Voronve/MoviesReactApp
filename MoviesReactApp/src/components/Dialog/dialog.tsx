import React from "react";
import { Portal } from "react-portal";
import './dialog.css';
import FocusTrap from 'focus-trap-react';
export interface dialogProps {
    title: JSX.Element | string,
    closeFunc: () => void,
    children: JSX.Element
}

/**
 * Custom dialog window
 */
export function Dialog( { title, closeFunc, children }: dialogProps ) {

    return (
        <Portal node={document.getElementById('dialogContainer')} >
            <div className={`dialogWindow`}>
                <div className="dialogHeader">{title}</div>
                {children}
                <button className="crossButton" onClick={() => closeFunc()}>X</button>
            </div>
        </Portal>
    )
}