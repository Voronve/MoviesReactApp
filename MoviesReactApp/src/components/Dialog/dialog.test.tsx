import { Dialog } from './dialog';
import { fireEvent, render, screen } from '@testing-library/react';

describe("Testing Dialog component", () => {
    const testTitle = <h2>Test title</h2>;
    const onSelect = jest.fn();
    const testChild = <div>Test child</div>;

    test("Testing dialog window rendering ", () => {

        const { asFragment } = render(<Dialog title={testTitle} closeFunc={onSelect}>{testChild}</Dialog>);

        expect(asFragment()).toMatchSnapshot();
    });

    test("Testing that dialog child props exists", () => {

        render(<Dialog title={testTitle} closeFunc={onSelect}>{testChild}</Dialog>);

        const title = screen.getByText("Test title");
        const child = screen.getByText("Test child");

        expect(title).toBeInTheDocument();
        expect(child).toBeInTheDocument();
    });

    test("Testing that closeFunc function in dialog is working as expected", () => {

        render(<Dialog title={testTitle} closeFunc={onSelect}>{testChild}</Dialog>);

        fireEvent.click(screen.getByRole('button'));

        expect(onSelect).toBeCalled();
    });
});