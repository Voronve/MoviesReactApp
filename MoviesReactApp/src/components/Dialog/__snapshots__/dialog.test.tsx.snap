// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Dialog component Testing dialog window rendering  1`] = `<DocumentFragment />`;
